kodi_addon_checker/config.py

```diff
@@ -0,0 +1,51 @@
+import json
+import os
+from argparse import ArgumentParser
+
+from kodi_addon_checker.report import ReportManager
+
+
+class Config(object):
+    def __init__(self, repo_path, cmd_args=None):
+        self.configs = {} if cmd_args is None else vars(cmd_args)
+        self._load_config(repo_path)
+
+    def _load_config(self, repo_path):
+        config_path = os.path.join(repo_path, '.tests-config.json')
+        if os.path.isfile(config_path):
+            with open(config_path) as json_data:
+                file_config = json.load(json_data)
+                if file_config is not None:
+                    for key, value in self.configs.items():
+                        if value is not None:
+                            file_config[key] = value
+                    self.configs = file_config
+
+    def is_enabled(self, value):
+        return self.configs.get(value, False)
+
+    def __getitem__(self, item):
+        return self.configs.get(item)
+
+
+class ConfigManager(object):
+    configurations = {}
+
+    @classmethod
+    def register(cls, config, description, default_value, action):
+        cls.configurations[config] = [description, default_value, action]
+
+    @classmethod
+    def fill_cmd_args(cls, parser: ArgumentParser):
+        # Add --reporter
+        parser.add_argument("--reporter", action="append", choices=list(ReportManager.reporters.keys()),
+                            help="""enable a reporter with the given name.
```

Comment:
We probably can't print the available reports here right? Can we add them manually at least then?

Reply:
We do in the `--help` message. Please check:
```
  --reporter {console}  enable a reporter with the given name. You can use
                        this option multiple times to enable more than one
                        reporters
```
where whatever comes in between `{...}` are the available options.